packages/jest-resolve/src/resolver.ts

```diff
@@ -526,6 +528,30 @@ export default class Resolver {
     return paths;
   }
 
+  getGlobalPaths(from: string, moduleName?: string): Array<string> {
+    if (!moduleName || moduleName[0] === '.') return [];
+
+    const cachedGlobalPath = this._globalPathCache.get(moduleName);
+    if (cachedGlobalPath) {
+      return cachedGlobalPath;
+    }
+
+    const modulePaths = this.getModulePaths(from);
+    const globalPaths: Array<string> = [];
+
+    const paths = require.resolve.paths(moduleName);
+    if (paths) {
```

Comment:
I may be missing something, but do we need to spend CPU time to grab the global paths every time? I'm thinking the solution could be as simple as grabbing those global dirs *once*, probably just straight away in `nodeModulesPaths.ts`:
```js
const GLOBAL_PATHS = findGlobalPaths();
function findGlobalPaths() {
  const paths=require.resolve.paths('/')
  // findIndex, slice, etc.
}
```
and append them to the paths list?

Reply:
LGTM

Reply:
I agree with grabbing only once, but it cannot be directly in `nodeModulesPaths.ts` since module paths will be cached. For instance, if the path without module name is cached. then it won't update

Reply:
Not sure I understand the caching problem. We're talking about the "global paths" that come after `/node_modules`. Aren't they independent of the `moduleName`, something like
```
[
  '/home/user/.node_modules',
  '/home/user/.node_libraries'
]
```